src/Units/Unit.java

```diff
@@ -187,28 +187,32 @@ public void fire(WeaponModel weapon)
 
   public int getHP()
   {
-    return (int) Math.ceil(HP);
+    return (int) Math.ceil(healthToHP(health));
   }
   public double getPreciseHP()
   {
-    return HP;
+    return healthToHP(health);
+  }
+  private static double healthToHP(int input)
+  {
+    return ((double)input)/10;
   }
 
   public void damageHP(double damage)
   {
-    HP -= damage;
-    if( HP < 0 )
+    health -= damage*10;
+    if( health < 0 )
     {
-      HP = 0;
+      health = 0;
     }
   }
   public double alterHP(int change)
   {
-    double before = HP;
-    // Change the unit's health, but don't grant more
-    // than 10 HP, and don't drop HP to zero.
-    HP = Math.max(0.1, Math.min(model.maxHP, getHP() + change));
-    return HP - before;
+    int before = health;
+    // Change the unit's health, but don't
+    // go over maxHP, and don't drop HP to zero.
```

Comment:
We should probably put this kind of info into javadoc comments so it's clear how `damageHP` and `alterHP` differ.